### PR TITLE
Fix javadocs for linter

### PR DESCRIPTION
**Pull request by nkramer44**
Signed-off-by: nkramer44 <noah.ph.kramer@gmail.com>

## High Level Overview of Change
Fix all the javadocs that were breaking the linter.

### Context of Change
Circle is failing because of recent changes to the javadoc linter.  This fixes all the things the linter was complaining about so other PRs can pass circle.

### Type of Change

- [x] Documentation Updates

## Before / After
Bad javadocs.
Good javadocs.

## Test Plan
N/A

### CODE REVIEW

**Comment by nkramer44** [comment]
@amiecorso I left a couple TODOs in here.  A bunch of the immutable interfaces have `@Nullable` fields.  This is fine, but I would opt for using `Optional` instead. Makes code much more readable but in declaration and at the call site. 

Sorry I didn't catch that in your previous PRs!

File: src/main/java/io/xpring/ilp/model/AccountBalance.java [hunk 1]
```diff
@@ -18,11 +18,16 @@
...
   /**
    * The accountId for this account balance.
+   *
+   * @return A {@link String} containing the accountId for this account balance.
    */
   String accountId();
 
   /**
-   * Currency code or other asset identifier that this account's balances will be denominated in.
+   * he currency code or other asset identifier that this account's balances will be denominated in.
```
Comment (theotherian) [inline src/main/java/io/xpring/ilp/model/AccountBalance.java:27]: Oh lawd he comin

Comment (nkramer44) [reply src/main/java/io/xpring/ilp/model/AccountBalance.java]: image

File: src/main/java/io/xpring/xrpl/model/XRPTransaction.java [hunk 2]
```diff
@@ -34,17 +36,24 @@
    * If provided, this transaction is only valid if the sending account's previously-sent transaction matches the
    * provided hash.
    * </p>
+   *
+   * @return A byte array containing the has value of another transaction.
```
Comment (theotherian) [inline src/main/java/io/xpring/xrpl/model/XRPTransaction.java:40]: Can haz hash value?

**Pull request merged**